nssp_demo/timeseries_forecasts.R

```diff
@@ -85,35 +154,87 @@ main <- function(model_run_dir, n_forecast_days = 28, n_samples = 2000) {
     select(date,
       ed_visits_target = Disease, ed_visits_other = Other,
       data_type
-    ) |>
-    as_tsibble(index = date)
-
-  forecast_other <- fit_and_forecast(target_and_other_data, n_forecast_days,
+    )
+  ## Time series forecasting
+  ## Fit and forecast other (non-target-disease) ED visits using a combination
+  ## ensemble model
+  forecast_other <- fit_and_forecast(
+    target_and_other_data,
+    n_forecast_days,
     n_samples,
-    target_col = "ed_visits_other", output_col = "other_ed_visits"
+    target_col = "ed_visits_other",
+    output_col = "other_ed_visits"
   )
-  forecast_baseline <- fit_and_forecast(target_and_other_data, n_forecast_days,
+  forecast_baseline_ts_count <- fit_and_forecast(
+    target_and_other_data,
+    n_forecast_days,
     n_samples,
     target_col = "ed_visits_target",
-    output_col = "baseline_ed_visits"
+    output_col = "baseline_ed_visit_count_forecast"
+  )
+  ## Generate CDC flat forecast for the target disease number of ED visits
+  forecast_baseline_cdc_count <- cdc_flat_forecast(
+    target_and_other_data,
+    target_col = "ed_visits_target",
+    output_col = "baseline_ed_visit_count_forecast",
+    data_frequency = "1 day",
+    aheads = 1:n_forecast_days
   )
 
-  save_path_other <- path(model_run_dir, "other_ed_visits_forecast",
-    ext = "parquet"
+  forecast_baseline_ts_prop <- forecast_baseline_ts_count |>
+    to_prop_forecast(forecast_other)
+
+  forecast_baseline_cdc_prop <- cdc_flat_forecast(
+    target_and_other_data |>
+      mutate(ed_visits_prop = ed_visits_target /
+        (ed_visits_target + ed_visits_other)),
+    target_col = "ed_visits_prop",
+    output_col = "baseline_ed_visit_prop_forecast",
+    data_frequency = "1 day",
+    aheads = 1:n_forecast_days
   )
-  save_path_baseline <- path(model_run_dir, "baseline_ed_visits_forecast",
-    ext = "parquet"
+
+  to_save <- dplyr::bind_rows(
+    list(
+      basename = "other_ed_visits_forecast",
+      value = forecast_other
+    ),
+    list(
+      basename = "baseline_ts_count_ed_visits_forecast",
+      value = forecast_baseline_ts_count
+    ),
+    list(
+      basename = "baseline_ts_prop_ed_visits_forecast",
+      value = forecast_baseline_ts_prop
+    ),
+    list(
+      basename = "baseline_cdc_count_ed_visits_forecast",
+      value = forecast_baseline_cdc_count
+    ),
+    list(
+      basename = "baseline_cdc_prop_ed_visits_forecast",
+      value = forecast_basline_cdc_prop
+    )
+  ) |>
```

Comment:
seems to be calling out for a `tribble`, but totally optional

Reply:
Done in 26b7c0d